lib/migrations/migrator.js

```diff
@@ -139,10 +139,10 @@ class Migrator {
   async migrate() {
     const version = await this.ldb.get(this.layout.V.encode());
 
-    if (version === null && this.migrateFlag !== -1)
-      throw new Error('Database does not exist.');
-
     if (version === null) {
+      if (this.migrateFlag !== -1)
```

Comment:
instead of making this warning, we should follow the example of the migrations and make different check and throw different error:
```js
  const lastID = this.getLastMigrationID();

  if (this.migrateFlag != -1 && this.migrateFlag != lastID)
    throw new Error(`Migrate flag ${this.migrateFlag} does not match last ID: ${lastID}`);

```
Similar to the check in non-fresh db at line 163. (below)

Reply:
Ok I kept the warning but still throw an error if the migrate flag is the wrong number, I think that makes sense.